### PR TITLE
feat: Enable File I/O APM by default

### LINKED ISSUE #2264: Enable File I/O APM per default
### Description

The feature is GA with https://github.com/getsentry/sentry-cocoa/issues/1587, but still disabled by default. 
https://github.com/getsentry/sentry-cocoa/blob/9231f608d4b6fe4ec24f878e3c669d45ab946721/Sources/Sentry/SentryOptions.m#L70

Before enabling it by default we should measure the overhead.

### PR DESCRIPTION
**Pull request by kevinrenskers**
## :scroll: Description

See title.

## :bulb: Motivation and Context

Closes #2264.

## :green_heart: How did you test it?

Unit test.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [ ] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

**Comment by kevinrenskers** [comment]
Are there any docs that need to be updated?

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This version adds a dependency on Swift.
 ### Features
 
 - Properly demangle Swift class name (#2162)
+- Enable File I/O APM per default (#2497)
```
[the remark this answers is not in the document]
Comment (kevinrenskers) [reply CHANGELOG.md:10]: > The changelog entry seems to be part of an already released section ## 8.0.0

Uhm what?

[the remark this answers is not in the document]
Comment (philipphofmann) [reply CHANGELOG.md:10]: @brustolin released a beta yesterday. For betas we always copy all the changes for this version. So basically, copy everything from the latest beta to the unreleased section, and you are good to go.

**Comment by philipphofmann** [comment]
> Are there any docs that need to be updated?

Yes, but we do that once 8.0.0 is GA. We will batch that together in one bigger PR.

**Review by philipphofmann — APPROVED** [review]
`m`: Please also update the code docs https://github.com/getsentry/sentry-cocoa/blob/fded9f8f3dbf08284ebf3f0560a56318583f0d2a/Sources/Sentry/Public/SentryOptions.h#L261-L265

LGTM

**Pull request merged**